### ISSUE TITLE
double-click handler not working on google docs anymore

### ISSUE THREAD

**Issue by balmas** [alpheios-project/alpheios-core#169]
It looks like Google Docs has changed its UI and now our Alpheios double-click handler isn't functioning anymore at all.

the mouseover handler *does* work however, so maybe we could detect if the user is on a google docs site and if so switch automatically to mouseover?

**Comment by balmas** [comment]
dependent upon #183  so that we can notify the users appropriately of the change in behavior

**Comment by irina060981** [comment]
Merged and closed

**Comment by balmas** [comment]
in the QA build with this change (qa.20200320360) the google docs test always returns positive no matter what page you're on.

**Comment by balmas** [comment]
Alpheios Components 3.3.0-qa.20200320403

**Comment by monzug** [comment]
I do get the message that mouse move event is enabled. however the lookup works funny. I did enter a greek word e.g. βουλεύω and initially it did generate the morphology pop-up but then it went back to the lexical query produced no results. have you seen this?

![Screen Shot 2020-03-23 at 5 47 27 PM](https://user-images.githubusercontent.com/41396793/77341621-45ca5580-6d2f-11ea-9ba9-6f5cf9031415.png)

**Comment by balmas** [comment]
hmm. no, I can't reproduce that. @irina060981 can you?

I know we do have more work to do to make the mouseover handler work smoothly, particularly as pertains to the other alpheios ui elements (e.g. mouseover on the panel or popup itself shouldn't trigger a lookup).  See also #144

**Comment by irina060981** [comment]
Yes, I could reproduce it - it fires from button, button label Lookup for me.
After I spent some time for testing, I could see that data-ignore rule doesn't work any more.
I tried to add it directly to label, button - and it gets exclude flag , but lookup still executes.

**Comment by irina060981** [comment]
So it needs more detailed and deep investigation, I could do this, or could first work with Greek input.
@balmas , what do you think?

**Comment by balmas** [comment]
Hi @irina060981 thanks. I think this one is higher priority than the greek input.

**Comment by irina060981** [comment]
ok - will work on it tommorrow :)

**Comment by irina060981** [comment]
By the way it also works on words from all google menus.
And we are not able to add here `alpheios-enabled`.
May be we should limit to google docs id - `docs-editor-container`, where we have our text?

**Comment by irina060981** [comment]
Also this problem is also reproducable on any site if we would turn on mouseover.

**Comment by balmas** [comment]
> By the way it also works on words from all google menus.
> And we are not able to add here `alpheios-enabled`.
> May be we should limit to google docs id - `docs-editor-container`, where we have our text?

Ugh. I hate to get so specific with the google docs site, because they could change it at any time.
Maybe we could mitigate that by making it configurable, which element to apply it to?

**Comment by irina060981** [comment]
I think it is a good way to do :)

will invetigate what are the poblems with alpheios-ignore first and then will add such specific configuration. it could be useful for other sites for advanced users

**Comment by irina060981** [comment]
Merged and closed

**Comment by balmas** [comment]
Alpheios Components 3.3.0-qa.20200410462

@monzug please see discussion in #229 -- there are new settings for this, which I'm not sure if are worded clearly enough for the user.  Would like your feedback. Let me know if it's not clear. Thanks!

**Comment by monzug** [comment]
1) when I use alpheios on google doc (mouseover or double click) I get two pop-ups. I did try with Chinese words and greek words and I got two pop-ups in both scenarios.
see screenshots

![chinese-doc](https://user-images.githubusercontent.com/41396793/80605533-3a8ad980-8a01-11ea-8211-a0dedd5fd4aa.JPG)

![two-pop-ups](https://user-images.githubusercontent.com/41396793/80605542-3ced3380-8a01-11ea-856a-fa02a04f7e81.JPG)

2) we have  Mouse Move (in the message it's MouseMove) and mouse over in word selection. this is confusing. is mouse move = to mouse over?
the limit by id is fixed for now. Do we really need to showGoogle docs container id? 
![mousemove](https://user-images.githubusercontent.com/41396793/80606770-af124800-8a02-11ea-941e-2d8d17bdc547.JPG)

**Comment by irina060981** [comment]
>when I use alpheios on google doc (mouseover or double click) I get two pop-ups. I did try with Chinese words and greek words and I got two pop-ups in both scenarios.
see screenshots

I was not able to reproduce it on my google doc test
@monzug , could you give me your test google doc - I would try to reproduce

>we have Mouse Move (in the message it's MouseMove) and mouse over in word selection. this is confusing. is mouse move = to mouse over?

they are equal from user's point and for our current code . We started our developing from mouseOver and now we use mouseMove - but we didn't update the setting title. Should be fixed, I believe.

>Do we really need to showGoogle docs container id?
It is printed because it could be changed by google team developers. And we would like to give a a user opportunity to fix it.

>also, when I checked Use Mouseover to Select Words (recommended for Chinese) the mousemove message is not longer on the pop-ups.

It is expected behaviour, we show that message - because we enable mouseMove regardless of user settings. Of course I could change it if it is confusing.

**Comment by balmas** [comment]
>when I use alpheios on google doc (mouseover or double click) I get two pop-ups. I did try with Chinese words and greek words and I got two pop-ups in both scenarios.
see screenshots
I was not able to reproduce it on my google doc test
@monzug , could you give me your test google doc - I would try to reproduce

I can't reproduce this either. Tried in Chrome and FF

**Comment by balmas** [comment]
>we have Mouse Move (in the message it's MouseMove) and mouse over in word selection. this is confusing. is mouse move = to mouse over?
they are equal from user's point and for our current code . We started our developing from mouseOver and now we use mouseMove - but we didn't update the setting title. Should be fixed, I believe.

Agree we should fix this. I think we should use "Mouse Over". What do you think @monzug ?

**Comment by balmas** [comment]
>Do we really need to showGoogle docs container id?
It is printed because it could be changed by google team developers. And we would like to give a a user opportunity to fix it.

Maybe we should have an "Advanced" tab on the Options and move this to there.  We want to make it possible for the user to change it because Google Docs could make changes on their side to use a different id, and so it could allow people to workaround problems with that before we can get a new release out. But that is the only scenario where it should be changed.  What do you both think?

**Comment by balmas** [comment]
>also, when I checked Use Mouseover to Select Words (recommended for Chinese) the mousemove message is not longer on the pop-ups.
It is expected behaviour, we show that message - because we enable mouseMove regardless of user settings. Of course I could change it if it is confusing.

Yes this should only show if we are changing the behavior that the user was expecting.

**Comment by monzug** [comment]
1) I can't reproduce it any longer. I did do quite some testing with Chinese yesterday, then I switch to the google docs and could not rid off the double pop-ups... will try again (and I give you my mixed language google doc below
2) I think Mouse Over sounds better too.
3) Have a greek text on Google docs, the message Mouse Over is enabled for google docs is generated. if the mouse over checkbox is checked, the message is not longer there. I am still in Google docs, so I believe it should be still generated. 
4) I did suggest the advanced option in issue #288

![mouse-login-greek](https://user-images.githubusercontent.com/41396793/80713853-1f33d300-8ac2-11ea-899a-e05fb3546e32.JPG)

**Comment by monzug** [comment]
I was not able to reproduce the double pop-ups... so sorry.

**Comment by balmas** [comment]
> I was not able to reproduce the double pop-ups... so sorry.

Let's make a separate issue for it and label it a test case so that we remember to keep an eye out for it.

**Comment by monzug** [comment]
yes, I agree. I noticed that in a previous comment I mentioned the two pop-ups too, so there is something that does trigger it.

**Comment by balmas** [comment]
I was wondering if it has to do with the state prior to switching to google docs. E.g. maybe if a lookup was already open?

**Comment by monzug** [comment]
I did try that. maybe some setting that is trigging it?

**Comment by balmas** [comment]
@monzug can this issue be closed now?  We have these follow-up issues:

#292 
#288 
#289

**Comment by monzug** [comment]
sure
